config.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 the generic-device-plugin authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/mitchellh/mapstructure"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/squat/generic-device-plugin/deviceplugin"
+)
+
+const defaultDomain = "squat.ai"
+
+// initConfig defines config flags, config file, and envs
+func initConfig() error {
+	cfgFile := flag.String("config-file", "", "Path to the config file.")
```

Comment:
Question for you: what do you think about using `--config` as the flag without the `-file`?

Reply:
I like it. Shorter is usually better imo